Update lr_scheduler.py to check the type of eta_min

ravikiranrao opened the pull request:
Add float assertion to `eta_min` parameter in `CosineAnnealingWarmRestarts`.

Fixes #87757

ravikiranrao committed:
Update lr_scheduler.py

Add float assertion to `eta_min` parameter in `CosineAnnealingWarmRestarts`.

janeyx99 commented on torch/optim/lr_scheduler.py:
```diff
@@ -1367,6 +1367,8 @@ def __init__(self, optimizer, T_0, T_mult=1, eta_min=0, last_epoch=-1, verbose=F
         if T_mult < 1 or not isinstance(T_mult, int):
             raise ValueError("Expected integer T_mult >= 1, but got {}".format(T_mult))
+        if not isinstance(eta_min, float):
+            raise ValueError("Exprected float, but got {}".format(eta_min))
```
```suggestion
            raise ValueError("Expected float eta_min, but got {} of type {}".format(eta_min, type(eta_min))
```
nit with spelling, and give a better error message if the type is string since it’d be hard to tell without printing the type if someone passes in “1e-5”.

ravikiranrao replied:
Ah! Sorry for the typo. Thanks for having `type` in the error message. This makes more sense. @janeyx99

janeyx99 reviewed (commented):
Thank you for the change! I left a comment—once addressed, this should be good to go!

ravikiranrao committed:
Update lr_scheduler.py

Fixed typo and print the `type` of eta_min, as suggested here: https://github.com/pytorch/pytorch/pull/97003#discussion_r1140235974

ravikiranrao commented:
> Thank you for the change! I left a comment—once addressed, this should be good to go!

Addressed @janeyx99

janeyx99 commented:
alright, approving CI. let’s ensure it’s all green—in the meantime, could you update the PR title to be more descriptive, such as mentioning that you’re adding error checking for eta_min type in lr scheduler?

janeyx99 commented:
CI failures are legit, looks like we’re missing a )

ravikiranrao committed:
Error checking assertion for eta_min in CosineAnnealingWarmRestarts learning rate scheduler

ravikiranrao commented:
> CI failures are legit, looks like we’re missing a )

Done.

ravikiranrao commented:
CI failures show that the assertion fails when `eta_min` is `int` type. Fixing this.

ravikiranrao committed:
update to check the type of eta_min

ravikiranrao commented:
cc @albanD.

janeyx99 reviewed (approved):
lgtm!

kit1980 commented:
@pytorchbot merge -r
